### PR TITLE
Correcting Documentation, adding asserts and disabling failing ci tests

### PR DESCRIPTION
**Pull request by Anipik**
- Adding Some More Asserts
- Correcting netcoreapp 3.0 documentation
- Disabling\correcting netcoreapp 3.0 tests

### CODE REVIEW

File: src/Microsoft.ML.CpuMath/SseIntrinsics.cs [hunk 1]
```diff
@@ -952,6 +954,7 @@ public static unsafe void ScaleAddU(float a, float b, Span<float> dst)
 
         public static unsafe void AddScaleU(float scale, ReadOnlySpan<float> src, Span<float> dst, int count)
         {
+            Contracts.Assert(count <= dst.Length);
```
Comment (eerhardt) [inline src/Microsoft.ML.CpuMath/SseIntrinsics.cs]: How about asserting `src.Length` as well?

File: src/Microsoft.ML.CpuMath/AvxIntrinsics.cs [hunk 2]
```diff
@@ -1718,6 +1725,7 @@ public static unsafe float MaxAbsDiffU(float mean, ReadOnlySpan<float> src)
 
         public static unsafe float DotU(ReadOnlySpan<float> src, ReadOnlySpan<float> dst, int count)
         {
+            Contracts.Assert(count <= src.Length);
```
Comment (eerhardt) [inline src/Microsoft.ML.CpuMath/AvxIntrinsics.cs]: How about asserting `dst.Length` as well?

File: test/Microsoft.ML.Predictor.Tests/TestPredictors.cs [hunk 3]
```diff
@@ -339,7 +339,7 @@ public void BinaryClassifierLogisticRegressionBinNormTest()
         /// <summary>
         ///A test for binary classifiers
         ///</summary>
-        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 output differs from Baseline
+        [ConditionalFact(typeof(BaseTestBaseline), nameof(BaseTestBaseline.LessThanNetCore30OrNotNetCoreAnd64BitProcess))] // x86 output differs from Baseline and flaky on netcore 3.0
         [TestCategory("Binary")]
         public void BinaryClassifierLogisticRegressionGaussianNormTest()
```
Comment (eerhardt) [inline test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: Are you adding these 2 tests to the list of tests to investigate?

Comment (Anipik) [reply test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: i will add the it to the list

**Comment by Anipik** [comment]
@Ivanidzo4ka can you please revive this one ?

**Pull request merged**